src/hooks/use-chat-context.tsx

```diff
@@ -0,0 +1,183 @@
+	const [ themesList, setThemesList ] = useState< Record< string, string[] > >( {} );
+	const numberOfSites = sites?.length || 0;
+	const sitePath = selectedSite?.path || '';
+	const sitePort = selectedSite?.port || '';
+	const siteThemeDetails = selectedSite?.themeDetails;
+
+	const availableEditors = Object.keys( installedApps ).filter( ( app ) => {
+		return installedApps[ app as keyof InstalledApps ];
+	} );
+
+	const fetchPluginList = useCallback( async ( path: string ) => {
+		const { stdout, stderr } = await getIpcApi().executeWPCLiInline( {
+			projectPath: path,
+			args: 'plugin list --format=json --status=active',
+		} );
+		if ( stderr ) {
+			return [];
+		}
+		return parseWpCliOutput( stdout, [] );
+	}, [] );
+
+	const fetchThemeList = useCallback( async ( path: string ) => {
+		const { stdout, stderr } = await getIpcApi().executeWPCLiInline( {
+			projectPath: path,
+			args: 'theme list --format=json',
+		} );
+		if ( stderr ) {
+			return [];
+		}
+		return parseWpCliOutput( stdout, [] );
+	}, [] );
+
+	useEffect( () => {
+		let isCurrent = true;
+		// Initial load. Prefetch all the plugins and themes for the sites.
+		const run = async () => {
+			const result = await Promise.all( [
+				fetchPluginList( sitePath ),
+				fetchThemeList( sitePath ),
+			] );
+			if ( isCurrent && selectedSite?.id ) {
+				setInitialLoad( ( prev ) => ( { ...prev, [ selectedSite.id ]: true } ) );
+				setPluginsList( ( prev ) => ( { ...prev, [ selectedSite.id ]: result[ 0 ] } ) );
+				setThemesList( ( prev ) => ( { ...prev, [ selectedSite.id ]: result[ 1 ] } ) );
+			}
+		};
+		if (
+			selectedSite &&
+			! loadingSites &&
+			! initialLoad[ selectedSite.id ] &&
+			isCurrent &&
+			! pluginsList[ selectedSite.id ] &&
+			! themesList[ selectedSite.id ]
+		) {
+			run();
+		}
+		return () => {
+			isCurrent = false;
+		};
+	}, [
+		fetchPluginList,
+		fetchThemeList,
+		initialLoad,
+		loadingSites,
+		pluginsList,
+		selectedSite,
+		sites,
+		themesList,
+		sitePath,
+	] );
+
+	useWindowListener( 'focus', async () => {
+		// When the window is focused, we need to kick off a request to refetch the theme details, if server is running.
+		if ( ! selectedSite?.id || selectedSite.running === false ) {
+			return;
+		}
+		const plugins = await fetchPluginList( sitePath );
+		const themes = await fetchThemeList( sitePath );
+		setPluginsList( ( prev ) => ( { ...prev, [ selectedSite.id ]: plugins } ) );
+		setThemesList( ( prev ) => ( { ...prev, [ selectedSite.id ]: themes } ) );
+	} );
+
+	const contextValue = useMemo( () => {
+		return {
+			numberOfSites,
+			themeList: selectedSite?.id ? themesList[ selectedSite.id ] || [] : [],
+			pluginList: selectedSite?.id ? pluginsList[ selectedSite.id ] || [] : [],
+			wpVersion,
+			phpVersion: selectedSite?.phpVersion ?? DEFAULT_PHP_VERSION,
+			currentURL: `http://localhost:${ sitePort }`,
+			themeName: siteThemeDetails?.name,
+			isBlockTheme: siteThemeDetails?.isBlockTheme,
```

Comment:
I tested changing the theme on a site and noticed that these values don't reflect the change when focusing the app. I think we'd need to use the `useThemeDetails` hook to keep them updated here. I checked and the property `themeDetails` from the site details only gets updated when starting/stopping a site.

Reply:
You are right! Thanks for the observation! It should be fixed, now!